CodegenProposal.md

```diff
@@ -0,0 +1,1474 @@
+    }
+    /// AllAnimals.AsPet.AsWarmBlooded    
+    struct AsWarmBlooded: TypeCase, HasFragments {
+      var species: String { data["species"] }
+      var humanName: String { data["humanName"] }
+      var favoriteToy: String { data["favoriteToy"] }
+      var bodyTemperature: Int { data["bodyTemperature"] }
+
+      struct Fragments: ResponseObject {
+        var warmBloodedDetails: WarmBloodedDetails { _toFragment() }
+      }
+    }
+  }
+
+  /// AllAnimals.AsWarmBlooded  
+  struct AsWarmBlooded: TypeCase, HasFragments {
+    var species: String { data["species"] }
+    var bodyTemperature: Int { data["bodyTemperature"] }
+
+    struct Fragments: ResponseObject {
+        var warmBloodedDetails: WarmBloodedDetails { _toFragment() }
+    }    
+  }
+}
+```
+
+# Alternatives & Suggestions
+
+## Generate Default Parameter Values for InputObject Default Values
+
+For fields with default values provided by the schema, we have decided to generate the fields as optional, but not include the default values in the generated code.
+
+```graphql
+input MyInput {
+  size: RelativeSize = SMALL
+}
+```
+```swift
+struct MyInput: InputObject {
+  public private(set) var dict: InputDict
+
+  init(size: GraphQLNullable<RelativeSize>) { ... }
+  
+  /// If `nil`, defaults to server-provided default value (.SMALL)
+  var size: GraphQLNullable<RelativeSize> { ... }
+}
+```
+An alternative approach is to provide the default value as a generated default argument.
+```swift
+struct MyInput: InputObject {
+  public private(set) var dict: InputDict
+
+  init(size: GraphQLNullable<RelativeSize> = .some(.SMALL)) { ... }
+  
+  /// If `nil`, defaults to server-provided default value (.SMALL)
+  var size: GraphQLNullable<RelativeSize> { ... }
+}
+```
+This however does not account for the fact that future changes to the default value of a field on an input type in a schema are considered to be backwards compatible. By generating the default value, we create a client that explicitly sends the value that *was* the default value when the type was generated – not necessarily the current default value of the server. In this case, the user could still explicitly pass `nil` to the initializer to indicate the intention to use the current default value as resolved by the server. However this is unclear at the call site and does not fall inline with the intentions of input coercion in the GraphQL Spec.
+
+For this reason, we have opted to not generate default values.
+
+## Concrete Subtypes as Enums
+
+[@designatedNerd’s initial proposal includes enums with associated values for subtypes.](https://github.com/apollographql/StarWarsiOSMk2/blob/master/StarWarsMark2/Queries/HeroTypeDependentAliasedFieldQueryMk2.swift#L43)
+
+These `SubType` enums thing works for concrete types, not interfaces (because a type could conform to multiple interfaces). We don't plan on generating all the Concrete types as data structures unless they are specifically enumerated (`... on Droid`)
```

Comment:
"These subtype enums thing" could use a little cleanup :)

Reply:
lol yes, it could. :)